stake-pool/program/src/processor.rs

```diff
@@ -1218,38 +1246,127 @@ impl Processor {
             return Err(ProgramError::InsufficientFunds);
         }
 
-        create_stake_account(
-            transient_stake_account_info.clone(),
-            transient_stake_account_signer_seeds,
-            system_program_info.clone(),
-        )?;
+        let maybe_split_from_account_info =
+            if let Some(ephemeral_stake_seed) = maybe_ephemeral_stake_seed {
+                let ephemeral_stake_account_info = maybe_ephemeral_stake_account_info.unwrap();
```

Comment:
presume you want to port trents suggestion here also

Reply:
Yeah, I'll merge the other one first, rebase, then make that change, thanks for spotting